src/index.ts

```diff
@@ -0,0 +1,46 @@
+import * as _ from 'lodash';
+import { ide } from "./gui/ide";
+import { doc } from "./doc/doc";
+import { runTests } from "./tests/runtests_gui";
+import { handleCanvas, handleTurtle } from "./gui/standalone";
+
+import "./css/ide.css";
+import "./css/tgui.css";
+import "./css/codemirror.css";
+import "./css/documentation.css";
+
+window.addEventListener("load", function(event)
+{
+	let container:any = document.getElementById("ide-container");
+	container.innerHTML = "";
+	let w:any = window;
+	if(typeof w.TScript !== "undefined"){
+		let TS = w.TScript;
+		ide.standalone = true;
+		ide.create(container);
+		ide.sourcecode.setValue(TS.code);
+		ide.prepare_run();
+		switch(TS.mode){
+			case 'canvas':
+				handleCanvas();
+				break;
+			case 'turtle':
+				handleTurtle();
+				break;
+		}
+		ide.interpreter.run();
+	}else{
+		switch(window.location.search.slice(1)){
+			case 'doc':
+				doc.create(container);
+				break;
+			case 'unittest':
+				runTests(container);
+				break;
```

Comment:
I'm not sure including the unit tests in `index.html` is a good idea. This currently adds 218 kilobytes to the bundle, which will be downloaded every time the IDE is used. This seems pretty unnecessary, since the unit tests are rarely run by users (have they ever been?).

Of course this could be solved by adding yet another bundle, but maybe it's not necessary at all to have users run unit tests, since we now have an automated test setup. I think it's more likely that users report an issue directly in the issue tracker before they find the instruction in the readme to run unit tests.

Reply:
I would agree no one runs those. I am not even so sure if browser tests are needed anyway, it seems unlikely, that any of the tests pass in the mocha but not on a browser.

Reply:
Browser test are not important to me. Testing in node should be fine. Users do not need the tests, and they should not be included in `index.html` and also not in `tscript.js`.